src/System.Memory/src/System/Buffers/ReadOnlyBuffer_helpers.cs

```diff
@@ -0,0 +1,305 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Runtime.CompilerServices;
+
+namespace System.Buffers
+{
+    public readonly partial struct ReadOnlyBuffer<T>
+    {
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        internal bool TryGetBuffer(SequencePosition start, SequencePosition end, out ReadOnlyMemory<T> data, out SequencePosition next)
+        {
+            if (start.Segment == null)
+            {
+                data = default;
```

Comment:
Can we initialize data and next to default right at the start?

Reply:
I would cause reassignment, right now every variable is assigned only once before return.